### PR TITLE
framework/bluetooth: Implement event callback logic

### PR DESCRIPTION
**Pull request by Wootak**
apps/bluetooth: Add bt_adapter_enable() calling logic

### CODE REVIEW

File: framework/src/bluetooth/adaptation/bt-adatation-adapter.c [hunk 1]
```diff
@@ -119,8 +137,33 @@ int bt_adapt_get_bonded_device_list(sq_queue_t *dev_list)
...
 {
+	struct bt_le_scan_param param = {
+		.type = BT_HCI_LE_SCAN_PASSIVE,
+		.filter_dup = BT_HCI_LE_SCAN_FILTER_DUP_DISABLE,
+		.interval = 0x010,
+		.window = 0x010,
+	};
+
+	BT_DBG("Enter\n");
+
```
Comment (dh79pyun) [inline framework/src/bluetooth/adaptation/bt-adatation-adapter.c]: In later, we should provide the error handling method such as "in progress" or "BT disable".
In this implemention, there is no return value for bt_le_scan_start. So never return "In progress" or "BT disable" error.
Please consider this in the later patchset.

Comment (Wootak) [reply framework/src/bluetooth/adaptation/bt-adatation-adapter.c]: Thank you. I will consider to provide return values such as 'In progress' and 'BT disabled' and so on.

**Comment by seinfra** [comment]
Target : [647228c66b4f495d2a00338220404a80a7e7aa7f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [647228c66b4f495d2a00338220404a80a7e7aa7f] - Code Rule Check OK.

File: apps/examples/bluetooth/Makefile [hunk 2]
```diff
@@ -20,6 +20,8 @@
 -include $(TOPDIR)/Make.defs
 include $(APPDIR)/Make.defs
 
+CFLAGS += -I$(TOPDIR)/../framework/include/bluetooth
```
Comment (pillip8282) [inline apps/examples/bluetooth/Makefile]: it's not recommended to add include path like this way.
you can refer include path by modifying source files
System contained framework/include path

File: os/include/tinyara/bluetooth/bluetooth.h [hunk 3]
```diff
@@ -40,6 +40,7 @@
  * Public Function Prototypes
  ****************************************************************************/
 
+#include <stdio.h>
```
Comment (pillip8282) [inline os/include/tinyara/bluetooth/bluetooth.h]: please include <tinyara/config.h> above stdio.h

File: apps/examples/bluetooth/Makefile (hunk 2 above)
Comment (Wootak) [reply apps/examples/bluetooth/Makefile]: Done.

File: os/include/tinyara/bluetooth/bluetooth.h (hunk 3 above)
Comment (Wootak) [reply os/include/tinyara/bluetooth/bluetooth.h]: Done. Thank you.

**Comment by seinfra** [comment]
Target : [48b539b3d3ee08947351a852f83b45c45acbdd38] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [48b539b3d3ee08947351a852f83b45c45acbdd38] - Code Rule Check OK.

**Pull request merged**